test/Pool.test.js

```diff
@@ -15,7 +15,7 @@ describe('Pool', function () {
   function createPool(script, options) {
     var pool = new Pool(script, options);
 
-    after(() => {
+    afterEach(() => {
```

Comment:
Can you change this back to `after(...)`? I think using `afterEach` will call `pool.terminate` on every pool after every test, but we only need to call it once per pool.

Reply:
Don't we want to terminate the pool after every test? since the instance is created in scope if individual test cases. If we run clean up using `after` then all registered hooks run after the entire `Pool` `describe` block finishes executing.  I could be misunderstanding the purpose of the hook, but it seems like we want to clean up after each test as per how each `pool` instance is scoped to a test case.

Reply:
If you create a global `count` in `Pool.test.js` and adjust the `afterEach` to increment and log the count, you'll see that `count` grows to 1801, whereas we only call `createPool()` 66 times. That is because we keep terminating all pools that have been created since the start after every test again and again.

```js
// global
let count = 0

// ...

afterEach(() => {
  count++
  console.log('count', count)
  return pool.terminate();
});
```

One solution is using `after` instead of `afterEach`. That has one downside: all created pools stay in memory until the end, so that costs more memory. 

An more neat solution is to change `createPool` to create a pool and then add that to a global list with created pools. Then, we create a single `afterEach` (outside of `createPool`) which calls `pool.terminate()` on every pool in the global array and then clear the array. That way we will terminate each created pool only once, and we terminate all created pools after every test. Shall we go for that? Ie:

```js
  // Creating pool with this function ensures that the pool is terminated
  // at the end of the test, which avoid hanging the test suite if terminate()
  // hadn't been called for some reason
  let createdPools = []
  function createPool(script, options) {
    const pool = new Pool(script, options);
    createdPools.push(pool);
    return pool;
  }

  afterEach(async () => {
    while (createdPools.length > 0) {
      await createdPools.shift().terminate();
    }
  });
```

Reply:
Ah, yes this makes sense without a aggergate like an `Array` to track pools seems like a good solution here. I think it is out of scope of this PR But will be a nice follow up.

Reply:
👍 I'll update the logic separately from this PR